entities/base_card.rb

```diff
@@ -0,0 +1,54 @@
+class BaseCard
+  attr_accessor :balance
+  attr_reader :type, :number
+
+  def initialize
+    @number = generate_card_number
+  end
+
```

Comment:
def number
  @number ||= generate_card number
end

Reply:
Did as you advised